src/aws/flb_aws_util.c

```diff
@@ -799,6 +796,27 @@ flb_sds_t flb_get_s3_key(const char *format, time_t time, const char *tag, char
     s3_key = tmp_key;
     tmp_key = NULL;
 
+    /* Find all occurences of $UUID and replace with a random string. */
+    random_alphanumeric = flb_sts_session_name();
+    if (!random_alphanumeric) {
+        goto error;
+    }
+    /* only use 8 chars of the random string */
+    random_alphanumeric[8] = '\0';
+    tmp_key = replace_uri_tokens(s3_key, RANDOM_STRING, random_alphanumeric);
+    if (!tmp_key) {
+        goto error;
```

Comment:
in the `goto error` case, will the `random_alphanumeric` variable be freed? I think it won't.

Reply:
yeah, I would free it before `goto error`.